src/v/kafka/server/handlers/topics/topic_utils.cc

```diff
@@ -48,13 +50,17 @@ ss::future<> wait_for_leaders(
             // topic already deleted
             continue;
         }
+        /**
+         * Partition space is contiguous, use integer range to iterate over all
+         * topic partitions.
+         */
+        int32_t partition_count = md->get().get_assignments().size();
```

Comment:
Could we not just call get_assignments() here and save it to a local? Or is the copy expensive?

Reply:
thats one per partition right? So if there are many partitions than thats a big copy?

Reply:
the copy is expensive, it is per partition

Reply:
OK, makes sense. As usual this stuff where we hold references across a scheduling point is a minefield. It does feel like maybe these cache methods should return shared pointers, so the caller keeps the metadata alive as long as it needs it.